### DISCUSSION TITLE
Table names with acronyms not generating properly

### DISCUSSION THREAD

**Discussion by jakeleventhal** [prisma/prisma#18205] [Q&A]
### Bug description

Acronyms don't follow proper camelcase rules. For instance, "PPC Campaign" is a table i want to create, so I create
```javascript
model PPCCampaign {...}
```

You would expect that the output would be `prisma.ppcCampaign` but instead, the output yield `prisma.pPCCampaign`.

Moreover, I tried adding a map to rename it:
```javascript
model ppcCampaign {
  id String @id @default(uuid())

  @@map("ppcCampaign")
}
```
but that doesn't seem to work.

I was able to solve this by naming the table `model ppcCampaign`, but this seems incorrect.

### How to reproduce

See above

### Expected behavior

Acronyms use camel case

### Prisma information

See above

### Environment & setup

- OS: macOS
- Database: PostgreSQL
- Node.js version: `v18.9.1`

### Prisma Version

```
4.10.1
```

**Comment by janpio** [comment]
> You would expect that the output would be `prisma.ppcCampaign` but instead, the output yield `prisma.pPCCampaign`.

The logic there is to lowercase the first character so `Post` becomes `post`. Our code can not be smart enough to handle you case - and not mishandle other cases then.

> Moreover, I tried adding a map to rename it:
> 
> ```js
> model ppcCampaign {
>   id String @id @default(uuid())
> 
>   @@map("ppcCampaign")
> }
> ```
> 
> but that doesn't seem to work.

If your table should be called `PPCCampaign` but you want to work with `ppcComaign` in code, you would need to use:

```prisma
model ppcCampaign {
  id String @id @default(uuid())

  @@map("PPCCampaign")
}
```

**Comment by jakeleventhal** [comment]
https://github.com/jsx-eslint/eslint-plugin-react/issues/3243

A similar issue was resolved in eslint. I think that acronyms are valid table names and prisma should support that.

**Reply by janpio** [reply]
Slightly different situation - our JS/TS API for 3+ years defaults to lowercasing the first character. Changing that behavior would be a breaking change for all of our users. If you want to be able to use a uppercase first letter in a model in the JS/TS API, open a feature request for it please. (We will most probably not go to implement this any time soon, but it could collect feedback from others that might have the same desire.)

**Reply by jakeleventhal** [reply]
The ask isn't to uppercase the first letter in a model. The ask is to skip capitalization of the next letter(s) when acronyms are used. In the jsx-eslint issue I linked for example, `btcValue` should not result in `bTCValue`. It should result in `btcValue`. Acronyms should either be inferred, or allowed via mapping.

**Reply by janpio** [reply]
Prisma let's you use any capitalization in the schema you want, both for Prisma internal names and for the mapped database names. Prisma generally never uppercases anything. It does lowercase the first letter for the JS/TS API. Changing that would be a breaking change, as I described above. Are you asking for something that would be possible without introducing that breaking change?

**Reply by aseemk** [reply]
@janpio: I just ran into this too and spent a while googling before finding this discussion. (Hard to google since so much content assumes you're asking to map the _DB table_, not the _Prisma client's field_ for the model.)

I strongly +1 @jakeleventhal: I'd like my model named `IAPTransaction` to generate to `prisma.iapTransaction`, not `prisma.iAPTransaction`. (IAP = "in-app purchase")

We're not asking to not lowercase the first letter. We're asking to _also lowercase_ the next letters if they're followed by another uppercase letter.

If you think even _that_ would be an _undesired_ breaking change (would be curious for examples why anyone _wouldn't_ want this), the ability to customize this generated name, e.g. via another attribute similar to `@@map`, would be really nice.

Thanks for your consideration!
